### DISCUSSION TITLE
Search Bar Position is Inconsistent

### DISCUSSION THREAD

**Discussion by MaggieAppleton** [eggheadio/egghead-next#370] [Ideas & Research]
On the homepage the search bar is in the navigation:
![Build the portfolio you need to be a badass web developer   egghead io](https://user-images.githubusercontent.com/5599295/106445596-1d46d800-6477-11eb-9738-81881941708d.png)

On the React page it has suddenly disappeared and jumped down into the page layout
![In-Depth Up-to-Date React Tutorials for 2021  egghead io](https://user-images.githubusercontent.com/5599295/106445597-1e780500-6477-11eb-85f7-68a4111cec1f.png)

It needs to be moved back to the navbar. See this: https://roamresearch.com/#/app/egghead/page/_mCVvC40I

**Comment by joelhooks** [comment]
this is intentional and won’t be moved to the header on the search page right now

we can reevaluate in the future

**Comment by MaggieAppleton** [comment]
We can certainly have two search bars - one in the navbar and one in the page itself. We just need to use the default text to let users know why they're different.
"Search for..." in the navbar
"Search for React content..." in the on-page search bar

That's an established design pattern:

![image](https://user-images.githubusercontent.com/5599295/106498230-ff00cc80-64b6-11eb-8ca0-8a957814324a.png)

It's confusing for learners to see the search bar there on one page, and have it disappear/move on another.

It makes it harder for them to find what they need, and breaks their understanding of where elements live on the site.

What if someone is on the React page and decides they want to search for Vue content instead?
It's multiple clicks back and forth for them to do it, rather than having the "escape hatch" of a navbar search element where they can jump to where they need to go 

---

Also it is _incredibly_ confusing that the minute someone types content into that search box, the _whole React page dissappears_ and is replaced by search content.

I start here:

![image](https://user-images.githubusercontent.com/5599295/106499149-20ae8380-64b8-11eb-91ce-a1afe83d3ae1.png)

Start to type "hooks" into the search bar:

![image](https://user-images.githubusercontent.com/5599295/106499190-2e640900-64b8-11eb-9812-e39bd50c4997.png)

Whoa where did the page go and where am I now??

Hit back button -> end up back on the homepage instead of on the React page where I started
![image](https://user-images.githubusercontent.com/5599295/106499205-3459ea00-64b8-11eb-93ac-c789135f8fd4.png)

😭

---

Canonical research from Nielson Norman Group: https://www.nngroup.com/articles/search-visible-and-simple/
- "The best designs offer a simple search box on the homepage and play down advanced search and scoping"
- **"Put the search box at the top of the page, usually in the right hand corner (though the left works almost as well)."**
- "you should make search available from every page on the site; you cannot predict where users will be when they decide they are lost"

**Comment by joelhooks** [comment]
This advanced well beyond the "we need to move the bar into the header" I've converted it to a discussion.

I'm all in favor of upgrading the search experience holistically, but it's obviously more nuanced than simple placement and the follow-up comment suggests that we need to be considering site search across the entire site.

These seem like great ideas that could be packaged into something that we could implement.
